dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/ValueMapFlattener.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.dynamodb;
+
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+public class ValueMapFlattener {
+
+    private static final Logger logger = LoggerFactory.getLogger(ValueMapFlattener.class);
+
+    private final String separator;
+    private final Predicate<String> indexFilter;
+    private final UnaryOperator<String> indexMapper;
+
+    public static class Builder {
+
+        private ValueMapFlattener valueMapFlattener;
+        private String separator = "";
+        private Predicate<String> indexFilter = new IndexFilterBuilder().doAllowAll().build();
+        private UnaryOperator<String> indexMapping;
+
+        public Builder withSeparator(String separator) {
+            this.separator = separator;
+            return this;  //By returning the builder each time, we can create a fluent interface.
+        }
+
+        public Builder withIndexFilter(Predicate<String> filter) {
+            this.indexFilter = filter;
+            return this;  //By returning the builder each time, we can create a fluent interface.
+        }
+
+        public Builder withIndexMapping(UnaryOperator<String> mapping) {
+            this.indexMapping = mapping;
+            return this;
+        }
+
+        public ValueMapFlattener build() {
+            valueMapFlattener = new ValueMapFlattener(separator, indexFilter, indexMapping);
+            return valueMapFlattener;
+        }
+    }
+
+    private ValueMapFlattener(String separator, Predicate<String> indexFilter, UnaryOperator<String> indexMapper) {
+        this.separator = separator;
+        this.indexFilter = indexFilter;
+        this.indexMapper = indexMapper;
+    }
+
+    /**
+     * Flattens a nested valuemap read from DynamoDB streamrecord.
+     * @param identifier of the original dynamoDB record
+     * @param valueMap Map containing the values associated with the record
+     * @return A document usable for indexing in elasticsearch
+     */
+    public Map<String, String> flattenValueMap(String identifier, Map<String, AttributeValue> valueMap) {
+        Map<String, String> flattenedPublication = new HashMap<>();
+        flattenedPublication.put("identifier", identifier);
+
+        flatten(flattenedPublication, "", valueMap);
+        return flattenedPublication;
+    }
+
+    private void flatten(Map<String, String> target, String prefix, Map<String, AttributeValue> valueMap) {
+        logger.trace("flatten: prefix={} values={}", prefix, valueMap);
+        valueMap.forEach((k, v) -> {
+            if (v != null) {
+                String key = addIndexPrefix(prefix, k);
```

Comment:
Refactor for better readbility. Are we sure that Elastic search does not support objects as types?

Reply:
The ValueMapFlattener has it's own task.. going to be changed in NP-1301. leaving it for now...